runtime/java/src/common/Util.java

```diff
@@ -394,20 +393,23 @@ private static void hackyhackyUnparseList(ConsCell c, StringBuilder sb) {
 	 * Returns the terminals from a parser.
 	 */
 	private static <ROOT> Object getTerminals(CopperParser<ROOT, CopperParserException> parser) {
-		Class<? extends CopperParser> parserClass = parser.getClass();
-		try {
-			Method getTokens = parserClass.getMethod("getTokens");
-			List<Terminal> tokens = (List<Terminal>) getTokens.invoke(parser);
-			return new Thunk<ConsCell>(() -> {
-				List<NTerminalDescriptor> tds = tokens
-					.stream()
-					.map(Util::terminalToTerminalDescriptor)
-					.collect(Collectors.toList());
-				return ConsCellCollection.fromList(tds);
-			});
-		} catch(Throwable t) {
-			throw new TraceException("Failed to reflect to getTokens()", t);
-		}
+		// Class<? extends CopperParser> parserClass = parser.getClass();
+		// try {
+		// 	Method getTokens = parserClass.getMethod("getTokens");
+		// 	List<Terminal> tokens = (List<Terminal>) getTokens.invoke(parser);
+		// 	return new Thunk<ConsCell>(() -> {
+		// 		List<NTerminalDescriptor> tds = tokens
+		// 			.stream()
+		// 			.map(Util::terminalToTerminalDescriptor)
+		// 			.collect(Collectors.toList());
+		// 		return ConsCellCollection.fromList(tds);
+		// 	});
+		// } catch(Throwable t) {
+		// 	throw new TraceException("Failed to reflect to getTokens()", t);
+		// }
+
+		return ConsCell.nil;
+		//TODO: FIXME
```

Comment:
I _believe_ this is only used by the never-good-and-probably-bitrotted-by-now monto-silver stuff, which is mostly removed anyway

Reply:
Uhh isn't this part of how we call every Copper parser?

Reply:
Yes, but nothing uses the `TerminalDescriptor`s; they'd need to be removed on the Silver side too, but I'm fairly sure they can be removed.

Reply:
Aren't they also used in the eclipse IDE?  I need to check though.  And wouldn't they probably also be useful in a LSP implementation?

Reply:
Dunno about Eclipse; I thought we were gonna pair tree-sitter + LSP, so we wouldn't need this for tree-sitter; if we aren't, then yeah I suppose it'd make sense to keep them around.

Reply:
This is a TODO because we are bizarrely (`git blame`: nathan :cowboy_hat_face: ) using reflection to call a private method on the parser. We should just update Copper to add `getTokens()` to the interface instead of doing this nonsense.

Reply:
(fwiw, all the CI tests pass with this changed to return `[]` so if we are using it in the IDE stuff that's the only place...)

Reply:
Oh, were you gonna kill this on the Silver side? otherwise, all lgtm

Reply:
Oh, were you gonna kill this on the Silver side? otherwise, all lgtm